module/Motion/LineTracing.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file   LineTracing.cpp
+ * @brief  ライントレース動作の中間クラス
+ * @author YKhm20020
+ */
+
+#include "LineTracing.h"
+using namespace std;
+
+LineTracing::LineTracing(double _targetDistance, int _targetBrightness, int _pwm,
+                         const PidGain& _gain, bool& _isLeftEdge)
+  : targetDistance(_targetDistance),
+    targetBrightness(_targetBrightness),
+    pwm(_pwm),
+    gain(_gain),
+    isLeftEdge(_isLeftEdge)
+{
+}
+
+void LineTracing::run()
+{
+  int currentPid = 0;
+  int sign = 0;
+  Pid pid(gain.kp, gain.ki, gain.kd, targetBrightness);
+
+  // 事前条件を判定する
+  if(!isMetPrecondition(pwm, targetDistance)) {
+    return;
+  }
+
+  // 左右で符号を変える
+  sign = isLeftEdge ? -1 : 1;
```

Comment:
```suggestion
  edgeSign = isLeftEdge ? -1 : 1;
```
signだけだと何に対するものなのかわかりにくいから変更してほしい